edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/EdgeHubConfigParser.cs

```diff
@@ -26,32 +29,68 @@ public EdgeHubConfigParser(RouteFactory routeFactory, BrokerPropertiesValidator
             this.validator = Preconditions.CheckNotNull(validator, nameof(validator));
         }
 
-        public EdgeHubConfig GetEdgeHubConfig(EdgeHubDesiredProperties desiredProperties)
+        public EdgeHubConfig GetEdgeHubConfig(string twinJson)
         {
-            Preconditions.CheckNotNull(desiredProperties, nameof(desiredProperties));
+            EdgeHubConfig edgeHubConfig;
 
-            ReadOnlyDictionary<string, RouteConfig> routes = ParseRoutes(desiredProperties, this.routeFactory);
+            var twinJObject = JObject.Parse(twinJson);
+            if (twinJObject.TryGetValue(Core.Constants.SchemaVersionKey, out JToken token))
+            {
+                Version version;
 
-            Option<BrokerConfig> brokerConfig = this.ParseBrokerConfig(desiredProperties.BrokerConfiguration);
+                try
+                {
+                    version = new Version(token.ToString());
+                }
+                catch (Exception e)
+                {
+                    throw new InvalidSchemaVersionException($"Error parsing schema version: {token}, Exception: {e.Message}");
+                }
 
-            return new EdgeHubConfig(
-                desiredProperties.SchemaVersion,
-                routes,
-                desiredProperties.StoreAndForwardConfiguration,
-                brokerConfig);
+                if (version == Core.Constants.SchemaVersion_1_0)
+                {
+                    var desiredProperties = JsonConvert.DeserializeObject<EdgeHubDesiredProperties_1_0>(twinJson);
+                    edgeHubConfig = this.GetEdgeHubConfig(desiredProperties);
+                }
+                else if (version == Core.Constants.SchemaVersion_1_1)
+                {
+                    var desiredProperties = JsonConvert.DeserializeObject<EdgeHubDesiredProperties_1_1>(twinJson);
+                    edgeHubConfig = this.GetEdgeHubConfig(desiredProperties);
+                }
+                else if (version == Core.Constants.SchemaVersion_1_2)
+                {
+                    var desiredProperties = JsonConvert.DeserializeObject<EdgeHubDesiredProperties_1_2>(twinJson);
+                    edgeHubConfig = this.GetEdgeHubConfig(desiredProperties);
+                }
```

Comment:
@varunpuranik if we want to take this change release/1.1, we'd need to remove this block and everything that references EdgeHubDesiredProperties_1_2 in this change.